lib/db.js

```diff
@@ -215,6 +215,23 @@ module.exports = function (
 
   // READ
 
+  DB.prototype.checkPassword = function (uid, verifyHash) {
+    verifyHash = Buffer(verifyHash).toString('hex')
+    log.trace({ op: 'DB.checkPassword', uid: uid, verifyHash: verifyHash })
+    return this.pool.post('/account/' + uid.toString('hex') + '/checkPassword',
+      {
+        'verifyHash': verifyHash
+      })
+      .then(
+        function (body) {
+          return uid.toString('hex') === body.uid
+        },
```

Comment:
I see you're retuning the `uid`, but I think (correct me if I mis-understood), if you're in this function then you know the password check was ok. In which case, can you just `return true` instead of this check since the `body.uid` should be `uid` anyway? Is this a double-check for something I missed?